src/Microsoft.TemplateEngine.Cli/Commands/TemplateCommand.cs

```diff
@@ -259,5 +275,177 @@ private Option CreateChoiceOption(CliTemplateParameter parameter, IEnumerable<st
             return option;
         }
 
+        private ParseArgument<T> GetParseArgument<T>(CliTemplateParameter parameter, Func<string?, (bool, T)> convert)
+        {
+            return (argumentResult) =>
+            {
+                if (argumentResult.Parent is not OptionResult or)
+                {
+                    throw new NotSupportedException("The method should be only used with option.");
+                }
+
+                if (argumentResult.Tokens.Count == 0)
+                {
+                    if (or.IsImplicit)
+                    {
+                        if (parameter.DefaultValue != null)
+                        {
+                            (bool parsed, T value) = convert(parameter.DefaultValue);
+                            if (parsed)
+                            {
+                                return value;
+                            }
```

Comment:
@jonsequitur do we need to have this chunk? 
I couldn't find a test case that leads to this code. When option is not specified, this code is not called at all.

Reply:
If you want the delegate to be called even when no option is specified, set the `isDefault` parameter to `true`.